Source/carthage/main.swift

```diff
@@ -22,7 +22,7 @@ guard ensureGitVersion().first()?.value == true else {
 
 if let remoteVersion = remoteVersion(), localVersion() < remoteVersion {
 	let formatting = ColorOptions.Formatting(true)
```

Comment:
I’d lean against styling `stderr` output — for `stdout` we’ve followed the convention of `git` and other that when piping or when `env[TERM] == 'DUMB'` [(see checks here)]( https://github.com/Carthage/Carthage/blob/5f0d04aa95e2946cd3ee454c21ee6d29058177d2/Source/carthage/Environment.swift#L23-L39), we don’t apply styling.

Additionally, we provide the `--color never` flag+argument, we should avoid violating that.

Probably sensible to not style, as we have with other `stderr` output.

Definitely makes sense to message this on `stderr` though, PR's good for me with this change, thanks!